### PR TITLE
Inspector Controls: Rephrase, polish, and make consistent color labels.

### PR DESCRIPTION
**Pull request by jasmussen**
## Description

Color panel labels are a bit of a mix across panels and even across web and mobile. This PR attempts to clean it up.

Before:

Screenshot 2021-03-22 at 08 17 21

Screenshot 2021-03-22 at 08 17 29

After:

Screenshot 2021-03-22 at 08 19 45
Screenshot 2021-03-22 at 08 20 58

The motivation here is that the "settings" part of "Color settings" is reduntant information, and that we use sentence case for labels everywhere else.

## Types of changes

- Color settings → Color
- Text Color → Text color
- Background Color → Background color
- Icon background color → Background color

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] I've tested my changes with keyboard and screen readers. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

**Review by aaronrobertshaw — COMMENTED** [review]
Nice!

Looks good although I have just one minor question regarding the change of `Icon background color` to `Background color` on the social links block. 

That color control is not actually setting the social links block's background color, only the background for the icons within it. Would it be better to leave it as `Icon background color` as it is more accurate?

Happy to defer to you if greater consistency outweighs accurately labelling what the control affects 🙂

**Comment by jasmussen** [comment]
That does actually make sense. Thanks.

**Comment by gziolo** [comment]
I'm thinking about how to generalize those labels so we could have a more structured approach when injecting UI controls with `supports` in the block definition. The most common causes are:
- text
- background
- link

There are exceptions like the Button block and Social Links block where we have quite a unique approach:
- icon's outer part (aka icon background color) and icon's inner part (aka icon color)
- buttons outer part (aka background color) and the text (aka text color)

I think it's a bit confusing in the Button block and it shouldn't use the name background since it's different from the background you know from the Group or Heading blocks. Should we use a consistent naming that would work for both those blocks and better express the intent like `Inner color` and `Outer color`? This way, we would be able to use 5 different types of color in `supports` that might cover 99% of cases. @mtias and @nosolosw, I would like to hear your thoughts as well.

**Comment by jasmussen** [comment]
Could we use "Item background"?

**Comment by gziolo** [comment]
> Could we use "Item background"?

Yes, `item background` and `item foreground` would work as well 👍🏻

**Comment by oandregal** [comment]
:wave: If the same label names in the [color hook](https://github.com/WordPress/gutenberg/pull/30075/files?file-filters%5B%5D=.js) can be the same in use for all the blocks (including social links) we won't need to implement https://github.com/WordPress/gutenberg/pull/29155 (it was created because in https://github.com/WordPress/gutenberg/issues/28913 we wanted to migrate some blocks -including social links- to use the block support mechanism instead of the custom controls and labels).

**Comment by gziolo** [comment]
@nosolosw, it might be as simple as extending what you can see for links in the Paragraph block:
https://github.com/WordPress/gutenberg/blob/85bf3a793c1c3ddff20e537f344bbeffa6b1cbe7/packages/block-library/src/paragraph/block.json#L33-L35

For the Social Links block you could have:

```json
"supports": {
    "color": {
        "foregroundItem": true,
        "backgroundItem": true
    }
}
```

and we would only need to extend the handling in the color hooks in the block editor package.

**Comment by jasmussen** [comment]
Feel free to push any changes to this PR if that's helpful.

**Comment by jasmussen** [comment]
We can also separate things out, as this PR touches a slew of other things.

**Comment by gziolo** [comment]
Let's agree on the direction first, I don't know if what I propose makes sense with everything else that already exists 😄

**Comment by oandregal** [comment]
@gziolo To make sure I follow, there're two things here:

1) For the labels, we don't need to rename anything in the block supports. If we can use the same labels in the hook and in the custom controls, we don't need anything else for https://github.com/WordPress/gutenberg/issues/28913

2) The other thing you seem to be hinting at is that we should also rename the names of the block supports for colors so that:

- `text` => `foregroundItem`
- `background` => `backgroundItem`

Is that correct? If we were to rename them, I'd think we'd want `text` to become `color` and `background` to be `backgroundColor`. The other block supports (fontSize, lineHeight, border, etc) have the same name of the CSS property they represent and I guess these should be the same. `gradients` represents `background`, thought, so I suppose it could be renamed as well although `gradients` is fine by me. Finaly, `link` is being moved out of block supports to become an element of the block as per https://github.com/WordPress/gutenberg/issues/29891

Does this help?

**Comment by gziolo** [comment]
@nosolosw, I don't think this is what I proposed:

> This way, we would be able to use 5 different types of color in supports that might cover 99% of cases.

I think that the existing default values are here to stay as they cover 90% of cases:
- text color
- background color
- link color

We could add more options, but it might be as well confusing for people using API.

> For the labels, we don't need to rename anything in the block supports. If we can use the same labels in the hook and in the custom controls, we don't need anything else for #28913

Yes, as discussed in other places, we shouldn't cover labels in the `theme.json` or `block.json`.

**Comment by oandregal** [comment]
I've closed https://github.com/WordPress/gutenberg/pull/29155#issuecomment-806583346 in favor of this one. Talked with Greg about this: it looks like the first step is just aligning the labels in use by the UI controls generated by the hooks and the custom labels. We won't offer the ability to have custom labels per block in the UI controls generated by the hooks at the moment.

**Comment by jasmussen** [comment]
@nosolosw Alright, I have to work on some other things at the moment, but feel free to entirely take over this branch if it's useful to you.

**Comment by jasmussen** [comment]
What are next steps here?

**Review by gziolo — APPROVED** [review]

File: packages/block-library/src/button/color-edit.js [hunk 1]
```diff
@@ -68,9 +68,7 @@ function ColorPanel( { settings, clientId, enableContrastChecking = true } ) {
 	const [ detectedBackgroundColor, setDetectedBackgroundColor ] = useState();
 	const [ detectedColor, setDetectedColor ] = useState();
 
-	const title = isWebPlatform
-		? __( 'Color settings' )
-		: __( 'Color Settings' );
```
Comment (gziolo) [inline packages/block-library/src/button/color-edit.js:73]: @hypest, is there any reason this differs in the mobile app. Are there tests that depend on it that need to get updated?

**Comment by jasmussen** [comment]
Thank you for the approval! Did the thoughts here have any impact on next steps? https://github.com/WordPress/gutenberg/pull/30075#issuecomment-803881903

**Comment by gziolo** [comment]
> Thank you for the approval! Did the thoughts here have any impact on next steps? [#30075 (comment)](https://github.com/WordPress/gutenberg/pull/30075#issuecomment-803881903)

I think we are fine. I prefer we land with a more general approach rather than trying to account for more distinct use cases.

We need to double-check with the mobile team while they picked a different label in the mobile app before proceeding. See https://github.com/WordPress/gutenberg/pull/30075#discussion_r613046983.

**Comment by jasmussen** [comment]
Cool. Feel free to press the green button when you feel it's time then, thanks!

File: packages/block-library/src/button/color-edit.js (hunk 1 above)
Comment (fluiddot) [reply packages/block-library/src/button/color-edit.js:73]: As far as I checked, we don't have any test that relies on this. I checked the [commit](https://github.com/WordPress/gutenberg/pull/22384/commits/285f05db8d85d5a32d5fc8ed7838792d04b3ad0b) that introduced this change and it doesn't look like the decision of using different values has to do beyond a naming convention. Being this said, I'd say it's safe to continue with this change 👍 .

@hypest do you agree?

Comment (hypest) [reply packages/block-library/src/button/color-edit.js:73]: Yeap, I agree. To be on the safe side I created [this gutenberg-mobile PR](https://github.com/wordpress-mobile/gutenberg-mobile/pull/3375) to run all tests from the native mobile side of things. Edit: native mobile tests are green 👍.

Comment (hypest) [reply packages/block-library/src/button/color-edit.js:73]: 👋 @iamthomasbishop , what's your take on the use of Sentence case here? Will it be a problem for the mobile apps? See [for example this diff](https://github.com/WordPress/gutenberg/pull/30075/files#diff-b32b988b074d8489b32cea966fa9a3e5b16ea22e0e365bdae2b79141a8cf4885R39).

Comment (jasmussen) [reply packages/block-library/src/button/color-edit.js:73]: Another way to think about it is: is title-case worth extra code across the entire project?

Comment (gziolo) [reply packages/block-library/src/button/color-edit.js:73]: In Polish you don't even have the choice, it's a sentence case on all websites, sometimes they use uppercase for highlights 😄

Comment (iamthomasbishop) [reply packages/block-library/src/button/color-edit.js:73]: > what's your take on the use of Sentence case here

@hypest We're slowly moving towards sentence-casing on mobile, so we can remove the mobile-specific code here.

Comment (hypest) [reply packages/block-library/src/button/color-edit.js:73]: ✅ all systems go then! I will go ahead and merge this PR, thank you all!

**Pull request merged**